src/libtriton/bindings/python/objects/pyTritonContext.cpp

```diff
@@ -2049,8 +2057,8 @@ namespace triton {
         }
 
         /* Free cb if exists */
-        if (PyMethod_Check(function))
-          Py_DECREF(cb);
+        /*if (PyMethod_Check(function))
+          Py_DECREF(cb);*/
```

Comment:
Can you explain this?

Reply:
The python callbacks are wrapped into lambda functions in addCallback.
There is no increment of the reference there so it should not be decremented here.
To see the bug (python crash): add a python class's method as callback, then remove it explicitly, then re-add it.
removeAllCallbacks does not decrement any reference, so this one is safe.

The only solution I see to properly own a reference would be to add a unique_pointer with a deleter that calls Py_DECREF in the lambda capture..

Reply:
Can we add a test case of the crash in `unittests` and add a new comment at line 2060 in `pyTritonContext.cpp` like `FIXME: Add a unique_pointer with a deleter that calls Py_DECREF in the lambda capture.`